src/transport/SecureSession.h

```diff
@@ -152,20 +152,16 @@ class SecureSession : public Session
     uint16_t GetLocalSessionId() const { return mLocalSessionId; }
     uint16_t GetPeerSessionId() const { return mPeerSessionId; }
 
-    // Should only be called for PASE sessions, which start with undefined fabric,
-    // to migrate to a newly commissioned fabric after successful
-    // OperationalCredentialsCluster::AddNOC
-    CHIP_ERROR NewFabric(FabricIndex fabricIndex)
+    // Called when AddNOC has gone through sufficient success that we need to switch the
+    // session to reflect a new fabric
+    CHIP_ERROR AdoptFabricIndex(FabricIndex fabricIndex)
     {
-#if 0
-        // TODO(#13711): this check won't work until the issue is addressed
-        if (mSecureSessionType == Type::kPASE)
+        // It's not legal to augment session type for non-CASE/PASE
+        if ((mSecureSessionType != Type::kPASE) && (mSecureSessionType != Type::kCASE))
         {
-            SetFabricIndex(fabricIndex);
+            return CHIP_ERROR_INVALID_ARGUMENT;
         }
-#else
         SetFabricIndex(fabricIndex);
```

Comment:
This should still only happen for PASE sessions, not CASE sessions....

Reply:
Because while CASE can add the new credentials, it should remain on the old fabric? I guess I'm not sure exactly how that flow works. For PASE, isn't it also fair to say, we should only be going from "undefined fabric index" to a "defined fabric index"? Interesting question... can you AddNOC three times during a PASE commissioning?

Reply:
I fixed it